docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1379,6 +1364,96 @@ As a Zephyr OS user I want to... ???
 
 [/SECTION]
 
+[SECTION]
+TITLE: Thread Synchronization
+
+[REQUIREMENT]
+UID: ZEP-99
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Counting Semaphore
+STATEMENT: >>>
+The system must implement a semaphore synchronization primitive for coordinating access to shared resources among multiple tasks.
+<<<
+USER_STORY: >>>
+As a software developer tasked with implementing resource management in our real-time operating system (RTOS), I want to integrate a semaphore synchronization primitive to facilitate coordinated access to shared resources among multiple tasks.
+<<<
+
```

Comment:
One space between the requirements, otherwise it will not compile. Please test locally the strictdoc command.